### PR TITLE
Hfa 110

### PR DESCRIPTION
**Pull request by MartinMiklovic**

### CODE REVIEW

File: src/app/components/search-header/search-header.component.ts [hunk 1]
```diff
@@ -13,18 +14,32 @@ export class SearchHeaderComponent implements OnInit {
...
       enumerationFilters: new FormGroup(this.getEnumerationFilterIds()),
     });
+    if (this._stateService.searchFormValue) {
+      this.searchForm.setValue(this._stateService.searchFormValue);
+    }
+  }
+
+  private initializeFormSubscription(): void {
     this.searchForm.valueChanges.subscribe(searchFormValue => {
+      console.log(searchFormValue);
```
Comment (RandomUserIK) [inline src/app/components/search-header/search-header.component.ts]: vymazat

File: src/app/public-view/components/blog-gallery/services/blog-gallery-state.service.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+})
+export class BlogGalleryStateService {
+
+  private _currentPage = 1;
+
+  get currentPage(): number {
+    return this._currentPage;
+  }
+
+  set currentPage(value: number) {
```
Comment (RandomUserIK) [inline src/app/public-view/components/blog-gallery/services/blog-gallery-state.service.ts:14]: return type je `void`

File: src/app/public-view/components/projects-gallery/services/projects-gallery-state.service.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+  private _currentPage = 1;
+  private _searchFormValue: any;
+
+
+  get currentPage(): number {
+    return this._currentPage;
+  }
+
+  set currentPage(value: number) {
```
Comment (RandomUserIK) [inline src/app/public-view/components/projects-gallery/services/projects-gallery-state.service.ts:16]: `void`

File: src/app/public-view/components/projects-gallery/services/projects-gallery-state.service.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    this._currentPage = value;
+  }
+
+  get searchFormValue(): any {
+    return this._searchFormValue;
+  }
+
+  set searchFormValue(value: any) {
```
Comment (RandomUserIK) [inline src/app/public-view/components/projects-gallery/services/projects-gallery-state.service.ts:24]: `void`

**Review by RandomUserIK — APPROVED** [review]

File: src/app/public-view/components/blog-gallery/services/blog-gallery-state.service.ts [hunk 5]
```diff
@@ -0,0 +1,18 @@
+import {Injectable} from '@angular/core';
+
+@Injectable({
+  providedIn: 'root'
```
Comment (trofoSK) [inline src/app/public-view/components/blog-gallery/services/blog-gallery-state.service.ts:4]: musi to byt v roote ?

File: src/app/public-view/components/projects-gallery/services/projects-gallery-state.service.ts [hunk 6]
```diff
@@ -0,0 +1,27 @@
+import {Injectable} from '@angular/core';
+
+@Injectable({
+  providedIn: 'root'
```
Comment (trofoSK) [inline src/app/public-view/components/projects-gallery/services/projects-gallery-state.service.ts:4]: To iste

File: src/app/public-view/components/blog-gallery/services/blog-gallery-state.service.ts (hunk 5 above)
Comment (MartinMiklovic) [reply src/app/public-view/components/blog-gallery/services/blog-gallery-state.service.ts:4]: ano pouziva to aj admin aj public cast

File: src/app/public-view/components/projects-gallery/services/projects-gallery-state.service.ts (hunk 6 above)
Comment (MartinMiklovic) [reply src/app/public-view/components/projects-gallery/services/projects-gallery-state.service.ts:4]: ano pouziva to aj admin aj public cast

File: src/app/public-view/components/blog-gallery/services/blog-gallery-state.service.ts (hunk 2 above)
Comment (MartinMiklovic) [reply src/app/public-view/components/blog-gallery/services/blog-gallery-state.service.ts:14]: A 'set' accessor cannot have a return type annotation

**Review by RandomUserIK — APPROVED** [review]

**Review by trofoSK — APPROVED** [review]

**Pull request merged**